plugins/backend/catalog-backend-module-aws/src/providers/AWSDynamoDbTableDataProvider.ts

```diff
@@ -0,0 +1,170 @@
+import { DynamoDBDocument } from '@aws-sdk/lib-dynamodb';
+import { merge } from 'lodash';
+import { mapColumnsToEntityValues } from '../utils/columnMapper';
+import * as winston from 'winston';
+
+export type ValueMapping = {
+  entityPath: string;
+  template?: string;
+};
+
+type DdbTableDataConfigOptions = {
+  tableName: string;
+  identifierColumn?: string;
+  columnValueMapping?: { [key: string]: ValueMapping };
+};
+
+/**
+ * Provides entities from AWS DynamoDB service.
+ */
+export class AWSDynamoDbTableDataProvider implements EntityProvider {
+  private readonly accountId: string;
+  private readonly roleArn: string;
+  private readonly tableDataConfig: DdbTableDataConfigOptions;
+  private readonly logger: winston.Logger;
+
+  private connection?: EntityProviderConnection;
+
+  static fromConfig(config: Config, options: { logger: winston.Logger }) {
+    return new AWSDynamoDbTableDataProvider(
+      config.getString('accountId'),
+      config.getString('roleArn'),
+      config.get<DdbTableDataConfigOptions>('dynamodbTableData'),
+      options.logger,
+    );
+  }
+
+  constructor(
+    accountId: string,
+    roleArn: string,
+    options: DdbTableDataConfigOptions,
+    logger: winston.Logger,
+  ) {
+    this.accountId = accountId;
+    this.roleArn = roleArn;
+    this.tableDataConfig = options;
+    this.logger = logger;
+  }
+
+  getProviderName(): string {
+    return `aws-dynamo-db-table-${this.accountId}-${this.tableDataConfig.tableName}`;
+  }
+
+  async connect(connection: EntityProviderConnection): Promise<void> {
+    this.connection = connection;
+  }
+
+  async run(): Promise<void> {
+    if (!this.connection) {
+      throw new Error('Not initialized');
+    }
+
+    const credentials = fromTemporaryCredentials({
+      params: { RoleArn: this.roleArn },
+    });
+
+    const ddb = DynamoDBDocument.from(
+      new DynamoDBClient({
+        credentials,
+      }),
+    );
+    const dynamoDBClient = new DynamoDB({ credentials });
+
+    const account = await new STS({ credentials }).getCallerIdentity({});
+
+    const defaultAnnotations: { [name: string]: string } = {
+      [ANNOTATION_LOCATION]: `${this.getProviderName()}:${this.roleArn}`,
+      [ANNOTATION_ORIGIN_LOCATION]: `${this.getProviderName()}:${this.roleArn}`,
+      'amazon.com/dynamodb-table-name': this.tableDataConfig.tableName,
+    };
+
+    if (account.Account) {
+      defaultAnnotations['amazon.com/account-id'] = account.Account;
+    }
+
+    this.logger.info(`Querying table ${this.tableDataConfig.tableName}`);
+
+    try {
+      const tableRows = await ddb.scan({
+        TableName: this.tableDataConfig.tableName,
+      });
+      const tableDescription = await dynamoDBClient.describeTable({
+        TableName: this.tableDataConfig.tableName,
+      });
+      const idColumn =
+        this.tableDataConfig.identifierColumn ||
+        tableDescription.Table?.KeySchema?.[0].AttributeName;
```

Comment:
Do we need to check if tableDescription is undefined here?

Reply:
The client will throw is it doesn't exist so type safety helps us here to trust that it is always present